intruder_alarm/src/singly/mod.rs

```diff
@@ -20,11 +20,14 @@ mod tests;
 // Public API types
 //-----------------------------------------------------------------------------
 //  List
-/// An intrusive singly-linked list.
+/// An intrusive singly-linked list mainly for stack operations.
```

Comment:
Honestly, we could probably change this line to just
```rust
/// An intrusive singly-linked stack.
```

If you think it makes sense, I'd be open to even just renaming this type to `Stack` rather than `List`, too; might express the intended use more clearly. I'd make that change in a separate PR though.

Reply:
OK. That makes much more sense (both, the line here and renaming to Stack).